File: 209-min-sub-array-len/solution.py
from typing import List
import math
class Solution1122:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        Sum = 0
        min_len = float('inf')

        for right in range(len(nums)):
            Sum += nums[right]
            while Sum >= target:
                min_len = math.min(min_len, right - left + 1)
                print("Find shorter len ", min_len)
                Sum -= nums[left]
                left += 1
        
        return 0 if min_len == float('inf') else min_len #三元運算子

print(Solution1122().minSubArrayLen(7, [2, 3, 1, 2, 4, 3]))

File: 209-min-sub-array-len/solution.go
/*
自己完成
*/
package main

import (
	"fmt"
	"math"
)

func minSubArrayLen(target int, nums []int) int {
    l, r := 0, 0
	sum, minLen := 0, math.MaxInt
	for r < len(nums){
		sum += nums[r]
		for sum >= target{
			minLen = min(minLen, r-l+1)
			sum -= nums[l]
			l++
		}
		r++
	}
	if minLen == math.MaxInt{
		return 0
	} else {
		return  minLen
	}
}

func main(){
	nums := []int{2,3,1,2,4,3}
	target := 7
	fmt.Println("Ans = ", minSubArrayLen(target, nums))
}
/*
代碼執行:
go run ./209-min-sub-array-len
*/